tests/e2e-tests/src/ai-core.test.ts

```diff
@@ -1,20 +1,141 @@
+      for (let i = 0; i < MAX_RETRIES; i++) {
+        const deploymentDetail = await DeploymentApi.deploymentGet(
+          deploymentId,
+          {},
+          { 'AI-Resource-Group': 'i745181' }
+        ).execute();
+        if (deploymentDetail.status === targetStatus) {
+          return deploymentDetail;
+        }
+        await new Promise(resolve => setTimeout(resolve, RETRY_INTERVAL));
+      }
+      throw new Error(
+        `Deployment ${deploymentId} did not reach ${targetStatus} status within the expected time. ` +
+          'Please manually stop and delete the deployment.'
+      );
+    }
+
+    const sanitizedState = (state: AiDeploymentList | undefined) => ({
+      ...state,
+      resources: state?.resources.map(resource =>
+        _.omit(resource, ['modifiedAt'])
+      )
+    });
+
+    beforeAll(async () => {
+      const queryResponse = await DeploymentApi.deploymentQuery(
+        {},
+        { 'AI-Resource-Group': 'i745181' }
+      ).execute();
+      expect(queryResponse).toBeDefined();
+      initialState = queryResponse;
+    });
+
+    it('should create a deployment and wait for it to run', async () => {
+      const createResponse = await DeploymentApi.deploymentCreate(
+        { configurationId: '8e2ff27f-d65e-48c4-9b21-1bb2709abfd1' },
+        { 'AI-Resource-Group': 'i745181' }
+      ).execute();
+
+      expect(createResponse).toBeDefined();
+      expect(createResponse.message).toBe('Deployment scheduled.');
+      expect(createResponse.id).toBeDefined();
+
+      const runningDeployment = await waitForDeploymentToReachStatus(
+        createResponse.id,
+        'RUNNING'
+      );
+      expect(runningDeployment.status).toBe('RUNNING');
+      expect(runningDeployment.deploymentUrl).toBeTruthy();
+      createdDeploymentId = runningDeployment.id;
+    }, 180000);
+
+    it('should modify the deployment to stop it', async () => {
+      expect(createdDeploymentId).toBeDefined();
+
+      const modifyResponse = await DeploymentApi.deploymentModify(
+        createdDeploymentId!,
+        { targetStatus: 'STOPPED' },
+        { 'AI-Resource-Group': 'i745181' }
+      ).execute();
+
+      expect(modifyResponse).toBeDefined();
+      expect(modifyResponse.message).toBe('Deployment modification scheduled');
+
+      const stoppedDeployment = await waitForDeploymentToReachStatus(
+        createdDeploymentId!,
+        'STOPPED'
+      );
+
+      expect(stoppedDeployment.status).toBe('STOPPED');
+    }, 180000);
+
+    it('should delete the deployment', async () => {
+      expect(createdDeploymentId).toBeDefined();
+
+      const deleteResponse = await DeploymentApi.deploymentDelete(
+        createdDeploymentId!,
+        { 'AI-Resource-Group': 'i745181' }
+      ).execute();
+
+      expect(deleteResponse).toBeDefined();
+      expect(deleteResponse.message).toBe('Deletion scheduled');
+    });
+
+    afterAll(async () => {
+      expect(createdDeploymentId).toBeDefined();
```

Comment:
[q] why is this assertion here? It is already asserted in the "modify" test..

Reply:
It is not technically required but If creating a deployment fails, then `createdDeploymentId` would be undefined. This assertion is present in every test after creating deployment to stop execution for these tests.